src/pro/main/ipc/handlers/local_agent/tools/read_logs.ts

```diff
@@ -0,0 +1,206 @@
+import { z } from "zod";
+import { ToolDefinition, AgentContext } from "./types";
+import { db } from "@/db";
+import { chats } from "@/db/schema";
+import { eq } from "drizzle-orm";
+import { getLogs } from "@/lib/log_store";
+import type { ConsoleEntry } from "@/ipc/ipc_types";
+
+const readLogsSchema = z.object({
+  timeWindow: z
+    .enum(["last-minute", "last-5-minutes", "last-hour", "all"])
+    .optional()
+    .describe("Time range to fetch logs from (default: last-5-minutes)"),
+
+  type: z
+    .enum([
+      "all",
+      "client",
+      "server",
+      "edge-function",
+      "network-requests",
+      "build-time",
+    ])
+    .optional()
+    .describe("Filter by log source type (default: all)"),
+
+  level: z
+    .enum(["all", "info", "warn", "error"])
+    .optional()
+    .describe("Filter by log level (default: all)"),
+
+  sourceName: z
```

Comment:
i would perhaps just leave this out because the AI can just search via searchTerm anyways (and this isn't always applicable, e.g. for non supabase apps)